src/js/components/TextInput/TextInput.js

```diff
@@ -258,6 +258,21 @@ const TextInput = forwardRef(
     const extraProps = {
       onSelect: handleTextSelect,
     };
+
+    const onEnterEventHandler = (event, handleForm) => {
+      closeDrop();
+      if (activeSuggestionIndex >= 0 && handleSuggestionSelect) {
+        // prevent submitting forms when choosing a suggestion
+        if (handleForm) {
+          event.preventDefault();
+          event.persist();
```

Comment:
I think we should be persisting all the time. And, it would be nice to not need `handleForm` and put the `event.preventDefault()` in the non-drop Keyboard only.

Reply:
Good comment. I thought so too and I tried to avoid this condition and the extra argument of `handleForm` on the function and make it generic for the two use cases, unfortunately, it didn't work well with the `showDrop` Keyboard, there was an issue with the selection of a keyboard item after scrolling. 
Let me take a second look since I tried it in the early stages of the PR and maybe the code was not fully mature for it at that time.

Reply:
So I was able to confirm that the functionality wouldn't work well with `event.preventDefault()` for the `showDrop` scenario.
Let's say we remove the `handleForm` condition so `event.preventDefault() && event.persist()` are always there, once you open the drop, use the scroll down -> hover on an item -> arrow from it up/down and click enter, the item should be selected but it's not which is an issue caused by the event actions.

Reply:
I was suggesting that `event.preventDefault()` be moved out of this function and only put in the Enter handler for the non-drop Keyboard.